service/src/sync_state.rs

```diff
@@ -46,6 +61,16 @@ pub(crate) fn request_keys<E: TlsRemoteAttestation>(
 	println!("key provisioning successfully performed");
 }
 
-fn get_worker_url_of_last_finalized_sidechain_block() -> String {
-	String::new()
+/// Returns the url of the last sidechain block author that has been stored
+/// in the parentchain state as "worker for shard".
+///
+/// Note: The sidechainblock author will only change whenever a new parentchain block is
+/// produced. And even then, it might be the same as the last block. So if several workers
+/// are started in a timely manner, they all will all get the same url.
+fn get_author_url_of_last_finalized_sidechain_block<NodeApi: PalletTeerexApi>(
```

Comment:
This method returns Ok() in any cases, correct? Why we then use Result and not the value directly?

Reply:
`let enclave = node_api.worker_for_shard(shard)?.ok_or(Error::EmptyValue)?;`
Might return an error value (`?`).

Reply:
And a `None` which I'm transforming into an error aswell, because without a value we can not continue.